packages/amplify-mqtt-server/src/client.ts

```diff
@@ -0,0 +1,440 @@
+import v4 from 'uuid';
+import * as retimer from 'retimer';
+import * as Connection from 'mqtt-connection';
+import * as steed from 'steed';
+
+import { MQTTServer } from './server';
+import { Logger } from 'pino';
+
+export class Client {
+  private logger: Logger;
+  private subscriptions = {};
+  nextId: number = 1;
+  inflight: {} = {};
+  inflightCounter: number = 0;
+  private keepAlive: number = 0;
+  private _lastDedupId: number = -1;
+  private _closed: boolean = false;
+  private _closing: boolean = false;
+  private timer;
+  private id;
+  private will;
+  private clean;
+  constructor(private connection: Connection, private server: MQTTServer) {
+    this.logger = server.logger;
+    this.setup();
+  }
+
+  private setup() {
+    const client = this.connection;
```

Comment:
Please think of renaming, calling something totally different then its type can be misleading when reading the code, because a question will be raised, how a connection become a client? IMHO it is either always a connection or always a client, but not both. It degrades the maintainability index, and in this case we've a ```client``` member in a ```Client``` class ;-)